hypothesis-python/src/hypothesis/extra/ghostwriter.py

```diff
@@ -452,54 +452,72 @@ def _guess_strategy_by_argname(name: str) -> st.SearchStrategy:
+        # this doesn't make a runtime difference, and it's much nicer for use-cases
+        # like `equivalent(numpy.add, operator.add)`.
+        return [
+            inspect.Parameter(name=name, kind=inspect.Parameter.POSITIONAL_ONLY)
+            for name in ascii_lowercase[: func.nin]  # type: ignore
+        ]
+    return []
+
+
 def _get_params(func: Callable) -> Dict[str, inspect.Parameter]:
     """Get non-vararg parameters of `func` as an ordered dict."""
     try:
         params = list(get_signature(func).parameters.values())
     except Exception:
-        if (
-            isinstance(func, (types.BuiltinFunctionType, types.BuiltinMethodType))
-            and hasattr(func, "__doc__")
-            and isinstance(func.__doc__, str)
-        ):
-            # inspect.signature doesn't work on all builtin functions or methods.
-            # In such cases, we can try to reconstruct simple signatures from the docstring.
-            match = re.match(rf"^{func.__name__}\((.+?)\)", func.__doc__)
-            if match is None:
-                raise
-            args = match.group(1).replace("[", "").replace("]", "")
-            params = []
-            # Even if the signature doesn't contain a /, we assume that arguments
-            # are positional-only until shown otherwise - the / is often omitted.
-            kind: inspect._ParameterKind = inspect.Parameter.POSITIONAL_ONLY
-            for arg in args.split(", "):
-                arg, *_ = arg.partition("=")
-                arg = arg.strip()
-                if arg == "/":
-                    kind = inspect.Parameter.POSITIONAL_OR_KEYWORD
-                    continue
-                if arg.startswith("*") or arg == "...":
-                    kind = inspect.Parameter.KEYWORD_ONLY
-                    continue  # we omit *varargs, if there are any
-                if _iskeyword(arg.lstrip("*")) or not arg.lstrip("*").isidentifier():
-                    break  # skip all subsequent params if this name is invalid
-                params.append(inspect.Parameter(name=arg, kind=kind))
-
-        elif _is_probably_ufunc(func):
-            # `inspect.signature` doesn't work on ufunc objects, but we can work out
-            # what the required parameters would look like if it did.
-            # Note that we use args named a, b, c... to match the `operator` module,
-            # rather than x1, x2, x3... like the Numpy docs.  Because they're pos-only
-            # this doesn't make a runtime difference, and it's much nicer for use-cases
-            # like `equivalent(numpy.add, operator.add)`.
-            params = [
-                inspect.Parameter(name=name, kind=inspect.Parameter.POSITIONAL_ONLY)
-                for name in ascii_lowercase[: func.nin]  # type: ignore
-            ]
+        if params := _get_params_ufunc(func):
+            pass
+        elif params := _get_params_builtin_fn(func):
+            pass
```

Comment:
is it possible for a function to correctly have 0 parameters here? if so this might unnecessarily fall back from `[]`; maybe we should return `None` in such cases from eg `_get_params_ufunc` and check against `None` instead of any falsey value.

Reply:
If a function has zero params, we can't feed it inputs with Hypothesis!  So in that case we end up dropping it, making `[]` and `None` similar enough for our purposes here.